libs/application/core/src/validation/validators.ts

```diff
@@ -10,18 +14,17 @@ interface SchemaValidationError {
 
 function populateError(
   currentError: SchemaValidationError | undefined,
-  newError: ZodError | undefined,
-  pathToError: string,
-): SchemaValidationError | undefined {
-  if (newError === undefined) {
-    return currentError
-  }
-
-  if (!currentError) {
-    return { [pathToError]: newError.errors[0].message }
+  newError: ZodSuberror[] | undefined,
+  pathToError?: string,
+) {
```

Comment:
Could we write this function like this: 

```
function populateError(
  currentError?: SchemaValidationError = {},
  newError: ZodSuberror[] | undefined,
  pathToError?: string,
) {
  let errorObject = {}
  newError?.forEach((element) => {
    errorObject = set(errorObject, pathToError || element.path, element.message)
  })
   return merge(currentError, errorObject)
}
```

Reply:
Another solution would be to change the order of variables into `merge` and have it like this `return merge(errorObject, currentError)`  
I think it would result in the same function and then we could skip the if check and returning error at the bottom of the function.